Source/WebCore/accessibility/AXCoreObject.cpp

```diff
@@ -1069,6 +1069,45 @@ unsigned AXCoreObject::blockquoteLevel() const
     return level;
 }
 
+unsigned AXCoreObject::headingLevel() const
+{
+    if (isHeading()) {
+        unsigned level = ariaLevel();
+        if (level > 0)
+            return level;
+    }
+
+    unsigned tagLevel = headingTagLevel();
+    if (tagLevel > 0)
+        return tagLevel;
+
+    return 0;
+}
+
+unsigned AXCoreObject::hierarchicalLevel() const
+{
+    int level = ariaLevel();
+    if (level > 0)
+        return level;
+
+    // Only tree item will calculate its level through the DOM currently.
+    if (roleValue() != AccessibilityRole::TreeItem)
+        return 0;
+
+    // Hierarchy leveling starts at 1, to match the aria-level spec.
+    // We measure tree hierarchy by the number of groups that the item is within.
+    level = 1;
+    for (RefPtr parent = parentObject(); parent; parent = parent->parentObject()) {
```

Comment:
I know this was written this way before, not something you did. But `parent` isn't a great name for this local variable, since it's only our parent in the first iteration, and every one after that it's not. `ancestor` and `ancestorRole` would be more accurate.